### PR TITLE
[TIMOB-23209] Improve handling of view animations on multiple presses

### PR DESCRIPTION
**Pull request by infosia**
[TIMOB-23209](https://jira.appcelerator.org/browse/TIMOB-23209)
[TIMOB-18750](https://jira.appcelerator.org/browse/TIMOB-18750)

Attempt to gain parity with iOS. Basic idea is to suppress multiple layout changes (size & position) done at a time of animation.

### Steps to reproduce

* Tap red box while box is moving.

### Expected

* The box doesn't reset its position while it's moving. 
* Warning message `[WARN] New layout set while view animating` is shown

```javascript
var win = Ti.UI.createWindow({ backgroundColor: 'green' });
var box = Ti.UI.createView({
    backgroundColor: 'red',
    left: 100,
    height: '100',
    width: '100'
});

var pin1 = Ti.UI.createView({
    backgroundColor: 'blue',
    left: 150,
    height: '140',
    width: '6'
});
var pin2 = Ti.UI.createView({
    backgroundColor: 'blue',
    left: 200,
    height: '140',
    width: '6'
});
var pin3 = Ti.UI.createView({
    backgroundColor: 'blue',
    left: 250,
    height: '140',
    width: '6'
});

win.add(pin1);
win.add(pin2);
win.add(pin3);
win.add(box);

var pos = 150;
box.addEventListener('click', function () {
    var a = Ti.UI.createAnimation({
        left: pos,
        duration: 2000,
    });
    a.addEventListener('complete', function () {
       pos += 50;
    });
    box.animate(a);
});
win.open();
```

### CODE REVIEW

**Comment by garymathews** [comment]
CR: **PASS**
FR: **PASS**

:+1:

Tests are failing due to `` Titanium.Network.HTTPClient.POST multipart/form-data containing Ti.Blob`` which is unrelated

**Comment by infosia** [comment]
test this please

**Comment by infosia** [comment]
Merged #734 manually. Seems to be working good 👍 

```javascript
var win = Ti.UI.createWindow({ backgroundColor: 'green' });
var box = Ti.UI.createView({
    backgroundColor: 'red',
    right: 50,
    height: '100',
    width: '100'
});

var pin1 = Ti.UI.createView({
    backgroundColor: 'blue',
    right: 50,
    height: '140',
    width: '6'
});
var pin2 = Ti.UI.createView({
    backgroundColor: 'blue',
    right: 100,
    height: '140',
    width: '6'
});
var pin3 = Ti.UI.createView({
    backgroundColor: 'blue',
    right: 150,
    height: '140',
    width: '6'
});

win.add(pin1);
win.add(pin2);
win.add(pin3);
win.add(box);

var pos = 100;
box.addEventListener('click', function () {
    var a = Ti.UI.createAnimation({
        bottom: pos,
        duration: 2000,
    });
    a.addEventListener('complete', function () {
        pos += 50;
    });
    box.animate(a);
});
win.open();
```

**Comment by garymathews** [comment]
###### TEST FAILURES
```
Windows8_1.Phone.Titanium.UI.View.TIMOB-20598
Windows8_1.Phone.Titanium.UI.View.animate (left)
Windows8_1.Phone.Titanium.UI.View.animate (top)
Windows8_1.Store.Titanium.UI.View.TIMOB-20598
Windows8_1.Store.Titanium.UI.View.animate (left)
Windows8_1.Store.Titanium.UI.View.animate (top)
```

I think we need to disable them, like we do for Windows 10 Store?

**Comment by garymathews** [comment]
build win 8.1

**Comment by infosia** [comment]
@garymathews 

> I think we need to disable them, like we do for Windows 10 Store?

I'm not quite sure. Why `animate` doesn't work on WP/Store 8.1?

**Comment by sgtcoolguy** [comment]
I see this in the console log:

```
6:59:05 [INFO] :   Started: animate (top)
06:59:05 [WARN] :   New layout set while view animating
06:59:10 [INFO] :   Started: animate (left)
06:59:10 [WARN] :   New layout set while view animating
06:59:16 [INFO] :   Started: TIMOB-20598
06:59:16 [WARN] :   New layout set while view animating
```

My guess is that it's "squashing" the animation when it shouldn't be.

File: Source/UI/src/WindowsViewLayoutDelegate.cpp [hunk 1]
```diff
@@ -209,74 +209,82 @@ namespace TitaniumWindows
...
 
 		void WindowsViewLayoutDelegate::animate(const std::shared_ptr<Titanium::UI::Animation>& animation, JSObject& callback, const JSObject& this_object) TITANIUM_NOEXCEPT
 		{
+			// TIMOB-23209: Multiple layout trasnformations (position or size) can't done at a time.
+			const auto hasLayoutTransform = animation->hasLayoutTransform();
+			if (hasLayoutTransform && is_transforming_layout__) {
+				TITANIUM_MODULE_LOG_WARN("New layout set while view animating");
+				return;
+			}
+			is_transforming_layout__ = hasLayoutTransform;
```
Comment (sgtcoolguy) [inline Source/UI/src/WindowsViewLayoutDelegate.cpp]: My guess is that maybe we need to set this after the is_loaded__ check? From the test output it looks like we somehow get into a state where the if guard above happens on a single animation that is supposed to fire off on a window open event callback.

**Comment by sgtcoolguy** [comment]
Merged in #736

**Pull request closed without merging**